packages/react/cypress/component/auto/form/AutoForm.cy.tsx

```diff
@@ -64,6 +64,26 @@ describeForEachAutoAdapter("AutoForm", ({ name, adapter: { AutoForm }, wrapper }
       });
   });
 
+  it("onFailure callback should return an error if the form submission fails", () => {
+    const onFailureSpy = cy.spy().as("onFailureSpy");
+    cy.mountWithWrapper(<AutoForm action={api.widget.alwaysThrowError} exclude={["gizmos"]} onFailure={onFailureSpy} />, wrapper);
```

Comment:
`alwaysThrowError` is a special model action that throws an error no matter what.